pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,640 @@
+        for arg_name, inner_dict in ids.items():
+            _jobs = {}
+
+            for par_idx in inner_dict:
+                _shifts = shifts.get(arg_name)
+                if _shifts is not None:
+                    _shifts = _shifts.get(par_idx)
+
+                # Determine spectrum and number of frequencies, discounting for 0
+                _spectrum = spectra[arg_name][par_idx]
+                _R = len(_spectrum) - 1
+                if _shifts is not None:
+                    # Check whether 0 is among the shifts
+                    if any(
+                        qml.math.isclose(_shifts, qml.math.zeros_like(_shifts), rtol=0, atol=atol)
+                    ):
+                        need_f0 = True
+                    # Check whether the shifts have the correct size
+                    if len(_shifts) != 2 * _R + 1:
+                        raise ValueError(
+                            f"The number of provided shifts ({len(_shifts)}) does not fit to the "
+                            f"number of frequencies (2R+1={2*_R+1}) for parameter {par_idx} in "
+                            f"argument {arg_name}."
+                        )
+                else:
+                    # If no shifts are given, f0 is needed always
+                    need_f0 = True
+
+                # Store job
+                if _R > 0:
+                    _jobs[par_idx] = {"shifts": _shifts, "spectrum": _spectrum}
+                else:
+                    # _R=0 belongs to a constant function
+                    _jobs[par_idx] = None
+
+            jobs[arg_name] = _jobs
+
+    else:
+        jobs = {}
+        need_f0 = True
+        if ids is None:
+            ids = {outer_key: inner_dict.keys() for outer_key, inner_dict in nums_frequency.items()}
+        elif isinstance(ids, str):
+            # ids only provides a single argument name but no parameter indices
+            ids = {ids: nums_frequency[ids].keys()}
+        elif not isinstance(ids, dict):
+            # ids only provides argument names but no parameter indices
+            ids = {_id: nums_frequency[_id].keys() for _id in ids}
+        recon_fn = _reconstruct_equ
+
+        for arg_name, inner_dict in ids.items():
+            _jobs = {}
+
+            for par_idx in inner_dict:
+                _num_frequency = nums_frequency[arg_name][par_idx]
+                # Store job; f0 missing
+                _jobs[par_idx] = {"num_frequency": _num_frequency} if _num_frequency > 0 else None
+
+            jobs[arg_name] = _jobs
+
+    return ids, recon_fn, jobs, need_f0
+
+
+def reconstruct(qnode, ids=None, nums_frequency=None, spectra=None, shifts=None):
+    r"""Reconstruct univariate restrictions of an expectation value QNode.
+    For common quantum gates, such restrictions are finite Fourier series with known
+    frequency spectra. Thus they may be reconstructed using Dirichlet kernels or
+    a non-uniform Fourier transform.
```

Comment:
Will the term 'restriction' be understood to casual readers? 🤔 Unfortunately I don't have any better suggestions

Reply:
I like the phrasing you used in another suggestion: Reconstruct an expectation value QNode along a single parameter dimension.